src/waveform/renderers/allshader/waveformrendermark.cpp

```diff
@@ -183,23 +183,39 @@ void allshader::WaveformRenderMark::setup(const QDomNode& node, const SkinContex
     connect(pWaveformWidgetFactory,
             &WaveformWidgetFactory::untilMarkShowBeatsChanged,
             this,
-            &WaveformRenderMark::setUntilMarkShowBeats);
+            &WaveformRenderMark::setUntilMarkShowBeats,
+            Qt::UniqueConnection);
```

Comment:
Can you explain why this needs to be a Qt::UniqueConnection? Can it happen that allshader::WaveformRenderMark::setup() is called twice? When does it happen. The connect call is probably misplaces here and belongs to the constructor?

Reply:
> Can it happen that allshader::WaveformRenderMark::setup() is called twice?

It should not but this felt like an extra safety to ensure there is no connection duplication. What's is the downside of using this flag? I assume it will come with a small performance downside due to the uniqueness checks, but since this is a one-off call, is there a reason why we shouldn't use it?

Reply:
This has more aspects. In this PR it comes along as a bug fix, raises the question why the flag is required here. 

We in the rest of Mixxx we use it only if required. So this case looks like it is required and it taks an effort to find out that it is not is it? 

The other question is, about the purpose of the setup function. Can some calls be moved to the constructor? Why is the delayed setup required?

Reply:
> Why is the delayed setup required?

This is a common pattern in the renderers, this is mostly done for consistency AFAIT, since the LegacySkinParser would use a `setup` call to finish the element setup, based of its definition.

> Can some calls be moved to the constructor? 

Looks like nothing prevent doing some in the ctor so I moved everything there